File: budget-backend/src/main/java/com/example/budget/model/Account.java
package com.example.budget.model;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name="accounts")
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private double currentBalance;

    @ManyToOne
    @JoinColumn(name="currency_id",nullable = false)
    private Currency currency;

    @ManyToOne
    @JoinColumn(name="user_id",nullable = false)
    private User user;




}

File: budget-frontend/src/app/shared/types/account.ts
import {ICurrency} from './currency';
import {IUser} from './user';

export interface IAccount {
  id: number;
  name: string;
  currentBalance: number;
  currencyDto: ICurrency;
  userDto: IUser;
}
// export class Account implements IAccount {
//   id: number;
//   name: string;
//   currentBalance: number;
//   currencyDto: Currency;
//   userDto: User;
// }
